pkg/cloudprovider/aws/instancetypes.go

```diff
@@ -125,7 +126,7 @@ func (p *InstanceTypeProvider) getInstanceTypeZones(ctx context.Context) (map[st
 }
 
 // getInstanceTypes retrieves all instance types from the ec2 DescribeInstanceTypes API using some opinionated filters
-func (p *InstanceTypeProvider) getInstanceTypes(ctx context.Context) (map[string]*InstanceType, error) {
+func (p *InstanceTypeProvider) getInstanceTypes(ctx context.Context, provider *v1alpha1.AWS) (map[string]*InstanceType, error) {
 	if cached, ok := p.cache.Get(InstanceTypesCacheKey); ok {
```

Comment:
This isn't correct I don't think.  There's a single cache entry for all instance types, but their Resources() vary by the provider.  Could probably write a test to verify and ensure this stays fixed.

Reply:
Nice catch, I believe you're right.  I'm thinking about hashing the provider spec to generate a unique cache key.   I think I have a test which can reproduce it as well.

Reply:
Solved this by hashing the provider and adding it to the key.